ui/plugins/pinboard.js

```diff
@@ -245,5 +249,36 @@ treeherder.controller('PinboardCtrl', [
         $scope.pinnedJobs = thPinboard.pinnedJobs;
         $scope.relatedBugs = thPinboard.relatedBugs;
 
+        var getRevisionTips = function(projectName, list) {
```

Comment:
Ahh, I see what's happening.  The way this is written, it's trying to populate that list before we have any data in the ``resultset_store``.  But we don't need that list before we have any resultsets loaded.  So it's just a matter of doing this call a the wrong time.

I also think we should be doing this in ``ui/plugins/controller.js`` instead of ``pinboard.js``.

If you remove your changes from ``pinboard.js`` and add/update this code in ``controller.js`` it should work:

        $scope.$watch('getCountPinnedJobs()', function(newVal, oldVal) {
            if (oldVal === 0 && newVal > 0) {
                $scope.isPinboardVisible = true;
                getRevisionTips($scope.repoName, $scope.originalTipList);
            }
        });

        $scope.originalTipList = [];
        var getRevisionTips = function(projectName, list) {
            list.splice(0, list.length);
            var rsArr = ThResultSetStore.getResultSetsArray(projectName);
            _.forEach(rsArr, rs => {
                list.push({
                    revision: rs.revision,
                    author: rs.author
                });
            });
        };

Note, you're updating the ``viewJob`` function that's already there.  Since that's where the pinboard is shown.

Would you mind playing with this and see how it goes?  Thanks!  :)